bl/policy/policy_factory.go

```diff
@@ -102,6 +123,27 @@ func populateRules(policyRules []cliClient.Rule, customRules []*cliClient.Custom
 	return rules, nil
 }
 
+func checkRuleSchemaExist(customRule *cliClient.CustomRule, identifier string) error {
+	if customRule.Schema == nil && customRule.JsonSchema == "" {
+		return fmt.Errorf("invalid policy file: rule identifier \"%s\" has no associated schema", identifier)
+	}
+	return nil
+}
+
+func validateRuleSchema(schema interface{}) error {
+	jsonContent, err := json.Marshal(schema)
+	if err != nil {
+		return err
+	}
+
+	schemaLoader := gojsonschema.NewStringLoader(string(jsonContent))
+	_, err = gojsonschema.NewSchemaLoader().Compile(schemaLoader)
+	if err != nil {
+		return fmt.Errorf("invalid policy file: rule schema error: %s", err.Error())
```

Comment:
We would like to add the rule identifier in the error message, this way the user will know in which custom rule the error occurred.